File: CodeWars/Python/7Kyu/TVRemote.py
# URL = https://www.codewars.com/kata/tv-remote/python

def tv_remote(word):
    remoteControlKeyboard= ["abcde123","fghij456","klmno789","pqrst.@0","uvwxyz_/"]

    current = [0,0]
    stepsRequired = []

    for letter in word:
        for index, line in enumerate(remoteControlKeyboard):
            if letter in line:
                stepsRequired.append(abs(current[0] - line.index(letter)) + abs(current[1] - index) + 1)
                current[0] = line.index(letter)
                current[1] = index
 
    return sum(stepsRequired)

print(tv_remote("your"))

File: CodeWars/JavaScript/7Kyu/TVRemote.js
// URL: https://www.codewars.com/kata/tv-remote/train/javascript

var tvRemote = function(word) {
  const remoteControlKeyboard= ["abcde123","fghij456","klmno789","pqrst.@0","uvwxyz_/"];

  let current = [0,0];
  let stepsRequired = [];

  for (i=0;i<word.length;i++){
    for (j=0;j<remoteControlKeyboard.length;j++){
      if (remoteControlKeyboard[j].includes(word[i])){
        stepsRequired.push((Math.abs(current[0] - remoteControlKeyboard[j].indexOf(word[i]))) + (Math.abs(current[1] - j)) + 1);
        current[0] = remoteControlKeyboard[j].indexOf(word[i]);
        current[1] = j;
      }
    }
  }
  return stepsRequired.reduce((a,b)=>a+b,0);

}
